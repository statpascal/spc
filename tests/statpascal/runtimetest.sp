program runtimetest;

var 
    x: vector of real;
    s: string;

begin
    x := intvec (1, 10);
    writeln ('sqrt (x): ', sqrt (x));
    writeln ('sin (x):  ', sin (x));
    writeln ('cos (x):  ', cos (x));
    writeln ('log (x):  ', log (x));
    randomize (0);
    writeln ('perm (10): ', randomperm (10));
    s := 'Hello, world';
    insert ('to the ', s, 7);
    writeln (s);
    writeln (length (s));
    writeln ('POS: ', pos ('b', 'abc'));
    writeln (copy (s, 1, 5));
    delete (s, 1, 5);
    writeln (s);
    x := realvec (0, 7, 0.2);
    writeln (sin (x) * sin (x) + cos (x) * cos (x))
end.
