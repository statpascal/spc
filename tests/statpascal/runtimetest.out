sqrt (x): 1 1.41421 1.73205 2 2.23607 2.44949 2.64575 2.82843 3 3.16228 
sin (x):  0.841471 0.909297 0.14112 -0.756802 -0.958924 -0.279415 0.656987 0.989358 0.412118 -0.544021 
cos (x):  0.540302 -0.416147 -0.989992 -0.653644 0.283662 0.96017 0.753902 -0.1455 -0.91113 -0.839072 
log (x):  0 0.693147 1.09861 1.38629 1.60944 1.79176 1.94591 2.07944 2.19722 2.30259 
perm (10): 2 1 7 3 5 4 8 9 10 6 
Hello,to the  world
19
POS: 2
Hello
,to the  world
1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 
